src/Trigger.h

```diff
@@ -65,7 +65,7 @@ class Trigger final : public Obj, public notifier::detail::Receiver {
     // in where the "timeout" parameter appears, and in making the "loc"
     // parameter optional. ("loc" is only used for internal logging when
     // debugging triggers.)
-    Trigger(std::shared_ptr<WhenInfo> wi, const IDSet& globals, std::vector<ValPtr> local_aggrs, double timeout,
+    Trigger(const std::shared_ptr<WhenInfo>& wi, const IDSet& globals, std::vector<ValPtr> local_aggrs, double timeout,
```

Comment:
Hmm, hmm. Looking at the Trigger constructor's body, could probably deprecate the shared_ptr version altogether and go for `Trigger(const WhenInfo& wi, ...)`. But no need to do that here.